File: C++/645. Set Mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& n) {
        map<int,int>m;
        vector<int>a;
        for(int i:n){
        m[i]++;
        if(m[i]==2)
        a.push_back(i);
        }
        for(int i=1;i<=n.size();i++){
            if(m[i]==0){
                a.push_back(i);
                break;
            }
        }
        return a;
    }
};

File: JAVA/645. Set Mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        int[] result = new int[2];
        for (int num : nums) {
            map.put(num, map.getOrDefault(num, 0) + 1);
            if (map.get(num) == 2) result[0] = num;
        }
        for (int i = 1; i <= nums.length; i++) {
            if (!map.containsKey(i)) {
                result[1] = i;
                break;
            }
        }
        return result;
    }
}
